### ISSUE TITLE
unable to use SRT stream receiver mode

### ISSUE THREAD

**Issue by hereisderek** [datarhei/restreamer#635]
**Describe the bug**
unable to use SRT stream receiver mode

create a new stream as screenshot
![image](https://github.com/datarhei/restreamer/assets/839795/a8cfdc3d-58b4-4a47-a5f1-6d4b8abd0267)

**Expected behavior**
able to setup as srt stream server 

**Screenshots**

**Additional context**
i'm running it behind a reverse proxy, binds to the default 8080 port. and i have tested that my domain can be successfully resolved to the ip, with port forwarding enabled. 

error log
```
ts=2023-10-26T09:30:41Z level=INFO component="SRT" msg="no publisher for this resource found" address=":6000" client="[::1]:48433" handler="CONNECT" resource="991c1be5-d76c-4ea7-9336-92d178dfe1ee.stream" status="NOTFOUND"
ts=2023-10-26T09:30:41Z level=INFO component="SRT" msg="no publisher for this resource found" address=":6000" client="127.0.0.1:41640" handler="CONNECT" resource="991c1be5-d76c-4ea7-9336-92d178dfe1ee.stream" status="NOTFOUND"
ts=2023-10-26T09:30:41Z level=INFO component="Process" msg="Failed" id="restreamer-ui:ingest:991c1be5-d76c-4ea7-9336-92d178dfe1ee_probe"
ts=2023-10-26T09:30:41Z level=INFO component="Process" msg="Stopped" id="restreamer-ui:ingest:991c1be5-d76c-4ea7-9336-92d178dfe1ee_probe"
```

```
docker inspect restreamer
[
    {
        "Id": "50a9b605db2111429cfda275e2fa934168a23698beca06f1d4e27fbc3dbeb2c6",
        "Created": "2023-10-26T08:21:54.574824625Z",
        "Path": "/core/bin/run.sh",
        "Args": [],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 390588,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2023-10-26T08:47:45.712485582Z",
            "FinishedAt": "2023-10-26T08:47:41.756440765Z"
        },
        "Image": "sha256:0b184b623d723758c75969dba63cc2af3826fd5c0a2417085de681a2b5ce09be",
        "ResolvConfPath": "/var/lib/docker/containers/50a9b605db2111429cfda275e2fa934168a23698beca06f1d4e27fbc3dbeb2c6/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/50a9b605db2111429cfda275e2fa934168a23698beca06f1d4e27fbc3dbeb2c6/hostname",
        "HostsPath": "/var/lib/docker/containers/50a9b605db2111429cfda275e2fa934168a23698beca06f1d4e27fbc3dbeb2c6/hosts",
        "LogPath": "/var/lib/docker/containers/50a9b605db2111429cfda275e2fa934168a23698beca06f1d4e27fbc3dbeb2c6/50a9b605db2111429cfda275e2fa934168a23698beca06f1d4e27fbc3dbeb2c6-json.log",
        "Name": "/restreamer",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": [
            "fb1c20ed2e17efcd2f0aa0958ae8f9843e1e412ef77dbc6257e8ca29c350d502"
        ],
        "HostConfig": {
            "Binds": [
                "/mnt:/mnt:rw",
                "/mnt/unionfs/downloads:/downloads:rw",
                "/opt/restreamer/config:/core/config:rw",
                "tmpfs:/tmp/hls:rw"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {
                    "max-file": "3",
                    "max-size": "10m"
                }
            },
            "NetworkMode": "saltbox",
            "PortBindings": {
                "1935/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "1935"
                    }
                ],
                "1936/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "1936"
                    }
                ],
                "6000/udp": [
                    {
                        "HostIp": "",
                        "HostPort": "6000"
                    }
                ],
                "8080/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "8080"
                    }
                ],
                "8081/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "8081"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "ConsoleSize": [
                0,
                0
            ],
            "CapAdd": [],
            "CapDrop": null,
            "CgroupnsMode": "private",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": [],
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": [],
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [
                {
                    "PathOnHost": "/dev/video0",
                    "PathInContainer": "/dev/video0",
                    "CgroupPermissions": "rwm"
                },
                {
                    "PathOnHost": "/dev/video1",
                    "PathInContainer": "/dev/video1",
                    "CgroupPermissions": "rwm"
                },
                {
                    "PathOnHost": "/dev/video2",
                    "PathInContainer": "/dev/video2",
                    "CgroupPermissions": "rwm"
                },
                {
                    "PathOnHost": "/dev/video3",
                    "PathInContainer": "/dev/video3",
                    "CgroupPermissions": "rwm"
                },
                {
                    "PathOnHost": "/dev/snd/",
                    "PathInContainer": "/dev/snd/",
                    "CgroupPermissions": "rwm"
                },
                {
                    "PathOnHost": "/dev/dri/",
                    "PathInContainer": "/dev/dri/",
                    "CgroupPermissions": "rwm"
                }
            ],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ],
            "Init": false
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/6a7d92c60fa3ee2f2690065d1ed607a918e633105a570ec8d060f6b2149b1219-init/diff:/var/lib/docker/overlay2/37864a41e11ad974e39ac5e8415c4b44034c1637038c20eaed8744821f04de27/diff:/var/lib/docker/overlay2/ce484a031af8762fcc8256e857b8954b28428625d14c9d583a7cd70011ed0aca/diff:/var/lib/docker/overlay2/27a279d20120de3ed5202820595c64e045ee5944daca6e5d1ff15bf23540648e/diff:/var/lib/docker/overlay2/18edee919a1e955ba3da2e78d5856377f6c373e1e5b8d74c3c5926ca4f05dc26/diff:/var/lib/docker/overlay2/e5eed40f0834887bf1f20ebeee41b5e910ac36adf6188185d823f971a2083446/diff:/var/lib/docker/overlay2/c34cd73173f2ad40eddff45764744c439af93bf329cf1d6b60d42cc3d558cdb1/diff:/var/lib/docker/overlay2/fe8ef1a28371696b6e9ae3c8652ee459872a28ac4a42de1a8f6bc024b9132f6c/diff:/var/lib/docker/overlay2/e5c7dfb32e07ebf84c233e65c9ada23567ac07e6c06ca3ce9d451a93df9246ee/diff:/var/lib/docker/overlay2/25ea7d91cca26570eb1d6a0f5dfb60c6c526f930158ecf4d3797b7e0620e8bd0/diff:/var/lib/docker/overlay2/f3e6458fb6740640c9cf641b89ec0426c61d2a89f7af107d4d90edee3c2a32b8/diff:/var/lib/docker/overlay2/e5ac409c356a11bccccd42a6b0a42af8281502cc04f1059c02609b1927caf0e1/diff:/var/lib/docker/overlay2/02df5ef57b690caa0141820f650a28b8af07cf34cf2b799743edee6e5fb9af59/diff",
                "MergedDir": "/var/lib/docker/overlay2/6a7d92c60fa3ee2f2690065d1ed607a918e633105a570ec8d060f6b2149b1219/merged",
                "UpperDir": "/var/lib/docker/overlay2/6a7d92c60fa3ee2f2690065d1ed607a918e633105a570ec8d060f6b2149b1219/diff",
                "WorkDir": "/var/lib/docker/overlay2/6a7d92c60fa3ee2f2690065d1ed607a918e633105a570ec8d060f6b2149b1219/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "volume",
                "Name": "959ab97e5b441f22c3845aae74a43282ebcfaca2fc695fe0c4090568c2aa4b6b",
                "Source": "/var/lib/docker/volumes/959ab97e5b441f22c3845aae74a43282ebcfaca2fc695fe0c4090568c2aa4b6b/_data",
                "Destination": "/core/data",
                "Driver": "local",
                "Mode": "",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/mnt",
                "Destination": "/mnt",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/mnt/unionfs/downloads",
                "Destination": "/downloads",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/restreamer/config",
                "Destination": "/core/config",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "volume",
                "Name": "tmpfs",
                "Source": "/var/lib/docker/volumes/tmpfs/_data",
                "Destination": "/tmp/hls",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            }
        ],
        "Config": {
            "Hostname": "restreamer",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "1935/tcp": {},
                "1936/tcp": {},
                "6000/udp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "TZ=Pacific/Auckland",
                "PUID=1000",
                "PGID=1000",
                "CORE_API_AUTH_ENABLE=true",
                "CORE_API_AUTH_USERNAME=user",
                "CORE_API_AUTH_PASSWORD=pass",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "NVIDIA_VISIBLE_DEVICES=all",
                "NVIDIA_DRIVER_CAPABILITIES=compute,video,utility",
                "CORE_CONFIGFILE=/core/config/config.json",
                "CORE_DB_DIR=/core/config",
                "CORE_ROUTER_UI_PATH=/core/ui",
                "CORE_STORAGE_DISK_DIR=/core/data"
            ],
            "Cmd": null,
            "Image": "datarhei/restreamer:vaapi-dev",
            "Volumes": {
                "/core/config": {},
                "/core/data": {}
            },
            "WorkingDir": "/core",
            "Entrypoint": [
                "/core/bin/run.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "com.github.saltbox.saltbox_managed": "true",
                "diun.enable": "true",
                "org.opencontainers.image.ref.name": "ubuntu",
                "org.opencontainers.image.version": "20.04",
                "traefik.enable": "true",
                "traefik.http.routers.restreamer-api-http.entrypoints": "web",
                "traefik.http.routers.restreamer-api-http.middlewares": "globalHeaders@file,redirect-to-https,gzip",
                "traefik.http.routers.restreamer-api-http.priority": "30",
                "traefik.http.routers.restreamer-api-http.rule": "(Host(`restreamer.domain.com`) && PathPrefix(`/api`,`/ui`))",
                "traefik.http.routers.restreamer-api-http.service": "restreamer",
                "traefik.http.routers.restreamer-api.entrypoints": "websecure",
                "traefik.http.routers.restreamer-api.middlewares": "globalHeaders@file,secureHeaders@file,gzip",
                "traefik.http.routers.restreamer-api.priority": "30",
                "traefik.http.routers.restreamer-api.rule": "(Host(`restreamer.domain.com`) && PathPrefix(`/api`,`/ui`))",
                "traefik.http.routers.restreamer-api.service": "restreamer",
                "traefik.http.routers.restreamer-api.tls.certresolver": "cfdns",
                "traefik.http.routers.restreamer-api.tls.options": "securetls@file",
                "traefik.http.routers.restreamer-http.entrypoints": "web",
                "traefik.http.routers.restreamer-http.middlewares": "globalHeaders@file,redirect-to-https,gzip",
                "traefik.http.routers.restreamer-http.priority": "20",
                "traefik.http.routers.restreamer-http.rule": "Host(`restreamer.domain.com`)",
                "traefik.http.routers.restreamer-http.service": "restreamer-http",
                "traefik.http.routers.restreamer.entrypoints": "websecure",
                "traefik.http.routers.restreamer.priority": "20",
                "traefik.http.routers.restreamer.rule": "Host(`restreamer.domain.com`)",
                "traefik.http.routers.restreamer.service": "restreamer",
                "traefik.http.routers.restreamer.tls.certresolver": "cfdns",
                "traefik.http.routers.restreamer.tls.options": "securetls@file",
                "traefik.http.services.restreamer-http.loadbalancer.server.port": "8080",
                "traefik.http.services.restreamer.loadbalancer.server.port": "8080"
            },
            "StopTimeout": 10
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "3bd7ce0c1cc382d3945d699441e9fcbe3bffe932316eb5fd88a0012dc9975901",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "1935/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "1935"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "1935"
                    }
                ],
                "1936/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "1936"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "1936"
                    }
                ],
                "6000/udp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "6000"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "6000"
                    }
                ],
                "8080/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "8080"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "8080"
                    }
                ],
                "8081/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "8081"
                    },
                    {
                        "HostIp": "::",
                        "HostPort": "8081"
                    }
                ],
                "8181/tcp": null
            },
            "SandboxKey": "/var/run/docker/netns/3bd7ce0c1cc3",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "saltbox": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "restreamer",
                        "50a9b605db21"
                    ],
                    "NetworkID": "65ef46e59c46a6ba77ac4927882d5e260ce3a4aedaaf413d86b67bb8b837a11c",
                    "EndpointID": "6e3cc06461570cb5d985744a03b8c6b47a0c4f48dfb6887d5c49e0e6f9b36ebc",
                    "Gateway": "172.19.0.1",
                    "IPAddress": "172.19.0.6",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "fd00:dead:beef::1",
                    "GlobalIPv6Address": "fd00:dead:beef::6",
                    "GlobalIPv6PrefixLen": 48,
                    "MacAddress": "02:42:ac:13:00:06",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

```
#runing from inside the container
root@restreamer:/core# ffmpeg -f lavfi -re -i testsrc -c:v libx264 -preset ultrafast -f mpegts "srt://localhost:6000?mode=caller&transtype=live&streamid=2325b127-0382-40d1-88b5-439a98c60e65,mode:publish"
ffmpeg version 5.1.3-datahrei Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
  configuration: --extra-version=datahrei --prefix=/usr/local --enable-nonfree --enable-gpl --enable-version3 --enable-postproc --enable-static --enable-openssl --enable-libxml2 --enable-vaapi --enable-libv4l2 --enable-libfreetype --enable-alsa --enable-libsrt --enable-libx264 --enable-libx265 --enable-libvpx --enable-libmp3lame --enable-libopus --enable-libvorbis --disable-ffplay --disable-ffprobe --disable-debug --disable-doc
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
Input #0, lavfi, from 'testsrc':
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Video: rawvideo (RGB[24] / 0x18424752), rgb24, 320x240 [SAR 1:1 DAR 4:3], 25 tbr, 25 tbn
ffmpeg.inputs:[{"url":"testsrc","format":"lavfi","index":0,"stream":0,"type":"video","codec":"rawvideo","coder":"rawvideo","bitrate_kbps":0,"duration_sec":0.000000,"language":"und","fps":0.000000,"pix_fmt":"rgb24","width":320,"height":240}]
08:52:41.664687/SRT:RcvQ:w1!W:SRT.cn: processAsyncConnectRequest: REJECT reported from HS processing: Peer rejected connection - not processing further
[srt @ 0x56302dcb4fc0] Connection to srt://localhost:6000?mode=caller&transtype=live&streamid=2325b127-0382-40d1-88b5-439a98c60e65,mode:publish failed (Input/output error), trying next address
08:52:41.674455/SRT:RcvQ:w2!W:SRT.cn: processAsyncConnectRequest: REJECT reported from HS processing: Peer rejected connection - not processing further
[srt @ 0x56302dcb4fc0] Connection to srt://localhost:6000?mode=caller&transtype=live&streamid=2325b127-0382-40d1-88b5-439a98c60e65,mode:publish failed: Input/output error
srt://localhost:6000?mode=caller&transtype=live&streamid=2325b127-0382-40d1-88b5-439a98c60e65,mode:publish: Input/output error
root@restreamer:/core#
```

## Business inquiries
**We provide support for commercial requirements with professional support, agile software development, and consulting.** If you have a commercial request, be it a bug or a feature enhancement, please contact us directly at support@datarhei.com.

**Comment by hereisderek** [comment]
it used to work.. i accidently deleted the source now i couldn't get it working again..

**Comment by hereisderek** [comment]
saw this error report https://github.com/datarhei/restreamer/issues/584 not sure if it's relevant.

**Comment by ioppermann** [comment]
@hereisderek Do you have a token defined for the SRT server? This must be included in the `streamid` e.g. with `streamid=2325b127-0382-40d1-88b5-439a98c60e65,mode:publish,token:...`

I don't think that's an UDP connection problem. In the error message it says "Peer rejected connection". This means that the server actively rejected the connection. Probably because of the missing/wrong token.

**Comment by hereisderek** [comment]
> @hereisderek Do you have a token defined for the SRT server? This must be included in the `streamid` e.g. with `streamid=2325b127-0382-40d1-88b5-439a98c60e65,mode:publish,token:...`
> 
> I don't think that's an UDP connection problem. In the error message it says "Peer rejected connection". This means that the server actively rejected the connection. Probably because of the missing/wrong token.

i previously when tested didn't have any token set for srt (so was for passphrase too), now i just went and set both, and the url became :

`srt://domain.com:6000?mode=caller&transtype=live&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13,mode:publish,token:token&passphrase=1234567890`

however im getting the same result with this setting

```
ffmpeg version 5.1.3-datarhei Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 11.2.1 (Alpine 11.2.1_git20220219) 20220219
  configuration: --extra-version=datarhei --prefix=/usr --extra-libs='-lpthread -lxml2 -lm -lz -lsupc++ -lstdc++ -lssl -lcrypto -lz -lc -ldl' --enable-nonfree --enable-gpl --enable-version3 --enable-postproc --enable-static --enable-openssl --enable-libxml2 --enable-libv4l2 --enable-v4l2_m2m --enable-libfreetype --enable-alsa --enable-libsrt --enable-libx264 --enable-libx265 --enable-libvpx --enable-libmp3lame --enable-libopus --enable-libvorbis --disable-ffplay --disable-debug --disable-doc --disable-shared
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
05:47:18.436046/SRT:RcvQ:w1!W:SRT.cn: processAsyncConnectRequest: REJECT reported from HS processing: Peer rejected connection - not processing further
[srt @ 0x7ff876700bc0] Connection to srt://localhost:6000?mode=caller&transtype=live&latency=20000&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream,mode:request,token:token&passphrase=1234567890 failed (I/O error), trying next address
05:47:18.453865/SRT:RcvQ:w2!W:SRT.cn: processAsyncConnectRequest: REJECT reported from HS processing: Peer rejected connection - not processing further
[srt @ 0x7ff876700bc0] Connection to srt://localhost:6000?mode=caller&transtype=live&latency=20000&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream,mode:request,token:token&passphrase=1234567890 failed: I/O error
srt://localhost:6000?mode=caller&transtype=live&latency=20000&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream,mode:request,token:token&passphrase=1234567890: I/O error
```

**Comment by XeroxDev** [comment]
I had the same issue. You've to manually edit the url and put ".stream" behind the `streamid`. In your case:
`streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13`
would become
`streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream`

aka:
`srt://domain.com:6000?mode=caller&transtype=live&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13,mode:publish,token:token&passphrase=1234567890`
gets
`srt://domain.com:6000?mode=caller&transtype=live&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream,mode:publish,token:token&passphrase=1234567890`

and then it worked (for me). Don't know if this is expected behavior (I guess not), but it works.

**Comment by 126dimas** [comment]
> I had the same issue. You've to manually edit the url and put ".stream" behind the `streamid`. In your case: `streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13` would become `streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream`
> 
> aka: `srt://domain.com:6000?mode=caller&transtype=live&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13,mode:publish,token:token&passphrase=1234567890` gets `srt://domain.com:6000?mode=caller&transtype=live&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream,mode:publish,token:token&passphrase=1234567890`
> 
> and then it worked (for me). Don't know if this is expected behavior (I guess not), but it works.

TYVM. it works))

**Comment by XeroxDev** [comment]
> > I had the same issue. You've to manually edit the url and put ".stream" behind the `streamid`. In your case: `streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13` would become `streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream`
> > aka: `srt://domain.com:6000?mode=caller&transtype=live&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13,mode:publish,token:token&passphrase=1234567890` gets `srt://domain.com:6000?mode=caller&transtype=live&streamid=ac7f9b4d-edff-4b30-ad16-9557b0fc9f13.stream,mode:publish,token:token&passphrase=1234567890`
> > and then it worked (for me). Don't know if this is expected behavior (I guess not), but it works.
> 
> TYVM. it works))

Your welcome!

Maybe @datarhei can convert this issue into a bug report, so we don't need to create an additional issue?

**Comment by ioppermann** [comment]
Yes, this is a bug in the UI where the incomplete URL is displayed.

**Comment by ioppermann** [comment]
Fixed in datarhei/restreamer-ui@f4c9fbe61aa51adc0f761d8af68737da56b80d99

**Comment by svenerbeck** [comment]
Hello @hereisderek 

We are closing your ticket https://github.com/datarhei/restreamer/issues/635.

This may be due to the following reasons:

- Problem/inquiry has been solved
- Ticket remained unanswered by you for a more extended period
- Problem was explained and handled in another ticket

You can reopen this ticket at any time!

Please only open related tickets once! Always answer/ask in the original ticket with the same issue!
